docs/source/daml/intro/5_Restrictions.rst

```diff
@@ -66,34 +66,36 @@ For example, the simple Iou in :ref:`simple_iou` allowed the no-op where the ``o
+
+Now let's look at how the ``Redeem`` choice is exercised in a script:
+
 .. literalinclude:: daml/daml-intro-5/daml/Restrictions.daml
   :language: daml
   :start-after: -- REDEEM_TEST_BEGIN
   :end-before: -- REDEEM_TEST_END
 
-There are quite a few new time-related functions from the ``DA.Time`` and ``DA.Date`` libraries here. Their names should be reasonably descriptive so how they work won't be covered here, but given that Daml assumes it is run in a distributed setting, we will still discuss time in Daml.
-
-There's also quite a lot going on inside the ``do`` block of the ``Redeem`` choice, with several uses of the ``<-`` operator. ``do`` blocks and ``<-`` deserve a proper explanation at this point.
+For the purposes of testing the ``Redeem`` choice, the above code sets and advances the ledger time with the ``setTime`` and ``passTime`` functions respectively. Exercising the choice should fail or should not fail depending on the day of week and the time of day. While that is straightforward, the issue of time on a Daml ledger is worthy of more discussion.
 
 Time on Daml Ledgers
 --------------------
 
-Each transaction on a Daml ledger has two timestamps called the *ledger time (LT)* and the *record time (RT)*. The ledger time is set by the participant, the record time is set by the ledger.
+Each transaction on a Daml ledger has two timestamps:  the *ledger time (LT)* and the *record time (RT)*.
 
-Each Daml ledger has a policy on the allowed difference between LT and RT called the *skew*. The participant has to take a good guess at what the record time will be. If it's too far off, the transaction will be rejected.
+**Ledger time (LT)** is the time associated with a transaction in the ledger model, as determined by the participant. It is the time of a transaction from a business and application perspective. When you call ``getTime``, it is the LT that is returned. The LT is used when reasoning about related transactions and commits. The LT can be compared with other LTs to guarantee model consistency. For example, LTs are used to enforce that no transaction depends on a contract that does not exist. This is the requirement known as "causal monotonicity."
 
-``getTime`` is an action that gets the LT from the ledger. In the above example, that time is taken apart into day of week and hour of day using standard library functions from ``DA.Date`` and ``DA.Time``. The hour of the day is checked to be in the range from 8 to 18.
+**Record time (RT)** is the time assigned by the persistence layer. It represents the time that the transaction is “physically” recorded. For example, “The backing database ledger has assigned the timestamp of such-and-such time to this transaction.” The only purpose of the RT is to ensure that transactions are being recorded in a timely manner.
```

Comment:
```suggestion
**Record time (RT)** is the time assigned by the persistence layer. It represents the time that the transaction is recorded. For example, “The backing database ledger has assigned the timestamp of such-and-such time to this transaction.” The only purpose of the RT is to ensure that transactions are being recorded in a timely manner.
```

Reply:
I actually think it is helpful to have "physically" in this case. The reason is that the **Ledger time** is the official timestamp as far as the ledger model is concerned. The **Record time** just happens to be the time that the transaction is "physically" recorded.

I'm certainly open to discussing however.

Reply:
it's ok with me either way